### PR TITLE
WT-13777 Merge pm-3441-feature-branch into develop

### PR DESCRIPTION
**Pull request by Sean04**
The feature branch includes the following commits:

8a078173b2 [WT-13783](https://jira.mongodb.org/browse/WT-13783) Add return value check in schema_plan.c (#11243)
d7032d96bc [WT-13772](https://jira.mongodb.org/secure/RapidBoard.jspa?rapidView=467) Remove use of WT_ITEM_ALIGNED in util_block (#11236)
6197ada6c5 [WT-12494](https://jira.mongodb.org/browse/WT-12494) Checkpoint target option: remove documentation and close old tickets (#10937)
d6bba5cc03 [WT-12493](https://jira.mongodb.org/browse/WT-12493) Checkpoint target option: remove feature (#10934)
4ea91dac1b [WT-12492](https://jira.mongodb.org/browse/WT-12492) [WT-12492](https://jira.mongodb.org/browse/WT-12492) Checkpoint target option: remove tests and examples (#10932)
604715ac35 [WT-12481](https://jira.mongodb.org/browse/WT-12481) Custom extractors: remove docs (#10922)
6a210f1606 [WT-12480](https://jira.mongodb.org/browse/WT-12480) Remove custom extractors (#10884)
9b4a22f144 [WT-12479](https://jira.mongodb.org/browse/WT-12479) Custom extractors: remove tests and examples (#10851)
1cefe0e4ac [WT-13286](https://jira.mongodb.org/browse/WT-13286) Remove string in string.ok and test join02 and 07 from evergreen (#10880)
f19ac55bec [WT-12502](https://jira.mongodb.org/browse/WT-12502) Removing direct io from documentation (#10809)
2676dd35f5 [WT-13227](https://jira.mongodb.org/browse/WT-13227) Remove directio adjacent buffer alignment code (#10836)
95c94e593f s_string -r
bd3fd0aead [WT-12471](https://jira.mongodb.org/browse/WT-12471) Remove log-based incremental backup documentation (#10802)
7552ff2046 [WT-12462](https://jira.mongodb.org/browse/WT-12462) Remove rename api docs and instances (#10355)
d2212251dc [WT-12460](https://jira.mongodb.org/browse/WT-12460) Remove session rename feature (#10306)
e09458dde6 [WT-12459](https://jira.mongodb.org/browse/WT-12459) Remove session rename tests and testing frameworks (#10282)
5f79c8ddde [WT-12470](https://jira.mongodb.org/browse/WT-12470) Remove log based incremental backup (#10441)
9f0c1aa6db [WT-12501](https://jira.mongodb.org/browse/WT-12501) Remove directio feature (#10430)
98dc6d7061 [WT-12488](https://jira.mongodb.org/browse/WT-12488) Session join API: remove documentation and close old tickets (#10432)
65a512d651 [WT-12487](https://jira.mongodb.org/browse/WT-12487) Session join API: remove feature including __wt_bloom_inmem_get() and __wt_bloom_intersection() (#10415)
baea7fcce0 [WT-12469](https://jira.mongodb.org/browse/WT-12469) Remove log-based incremental backup tests/examples (#10405)
8514964c49 [WT-12486](https://jira.mongodb.org/browse/WT-12486) Session join API: Remove tests examples (#10372)
66e50561ab [WT-12500](https://jira.mongodb.org/browse/WT-12500) Remove direct_io tests and examples (#10387)
39c99e3f9a [WT-12472](https://jira.mongodb.org/browse/WT-12472) remove projections tests and examples (#10388)

### CODE REVIEW

**Review by lukech — APPROVED** [review]
LGTM! 
It's worth mentioning the set of tests we ran to prepare for the branch merge in the PR template comment.

**Comment by luke-pearson** [comment]
As I understand all these features being removed were reviewed prior to merge into the feature branch? 

> Please focus on ensuring this commit is accurate and includes the suitable information

Accurate is a funny word to use here, "suitable information" too. What does that mean exactly? I think the commit looks good, and suitable. I like how much code it is removing. 

This line in the code coverage report surprises me so I'll go track it down:  > In src/cursor/cur_backup.c the complexity of __backup_config has increased by 1 to 36.

**Comment by luke-pearson** [comment]
Can you point me to the commit that did this: 
```
#define WT_DIRECT_IO_CHECKPOINT 0x1ull /* Checkpoints */
#define WT_DIRECT_IO_DATA 0x2ull       /* Data files */
#define WT_DIRECT_IO_LOG 0x4ull        /* Log files */
                                       /* AUTOMATIC FLAG VALUE GENERATION STOP 64 */
    uint64_t direct_io;                /* O_DIRECT, FILE_FLAG_NO_BUFFERING */
    uint64_t write_through;            /* FILE_FLAG_WRITE_THROUGH */
#define WT_FILE_TYPE_DATA 0x1ull /* Data files */
#define WT_FILE_TYPE_LOG 0x2ull  /* Log files */
                                 /* AUTOMATIC FLAG VALUE GENERATION STOP 64 */
    uint64_t write_through;      /* FILE_FLAG_WRITE_THROUGH */
```

File: src/reconcile/rec_write.c [hunk 1]
```diff
@@ -585,10 +585,6 @@ __rec_init(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags, WT_SALVAGE_COO
         /* Connect pointers/buffers. */
         r->cur = &r->_cur;
         r->last = &r->_last;
-
-        /* Disk buffers need to be aligned for writing. */
```
Comment (luke-pearson) [inline src/reconcile/rec_write.c:589]: So are we always writing aligned or unaligned?

**Comment by Sean04** [comment]
> As I understand all these features being removed were reviewed prior to merge into the feature branch?
That's correct, if you want to look at a specific PR, they can be accessed through the PR number link (#X) in the description.

> Please focus on ensuring this commit is accurate and includes the suitable information.
Accurate in that I need a second pair of eyes to ensure we've included all of the committed tickets from the project and that there are no strange commits. Also, since this PR includes a group of changes I want to ensure the changes be easily traced in the git history. The description of this PR should contain the correct information so that each individual ticket can be traced. Do you think the descritpion/commit message should contain anything else?

**Comment by Sean04** [comment]
> Can you point me to the commit that did this:

https://github.com/wiredtiger/wiredtiger/commit/9f0c1aa6dbf120aa3f5c344e5ff79e31dd09293e [WT-12501](https://jira.mongodb.org/browse/WT-12501) Remove directio feature (https://github.com/wiredtiger/wiredtiger/pull/10430)

File: src/meta/meta_ckpt.c [hunk 2]
```diff
@@ -1452,9 +1451,9 @@ __wt_meta_sysinfo_set(WT_SESSION_IMPL *session, bool full, const char *name, siz
 
     /*
      * Write the checkpoint timestamp and snapshot information to the metadata. For any full
-     * checkpoint, including most named checkpoints, write the most recent checkpoint's entries. For
-     * all named checkpoints, whether or not full, write it to that checkpoint's entries by name.
-     * This writes out two copies for most named checkpoints, but that's ok.
+     * checkpoint, i.e. all checkpoints, write the most recent checkpoint's entries. For
```
[the remark this answers is not in the document]
Comment (Sean04) [reply src/meta/meta_ckpt.c:1454]: https://github.com/wiredtiger/wiredtiger/commit/d6bba5cc03e9d609b63e29dca3f60aba2c14b3aa [WT-12493](https://jira.mongodb.org/browse/WT-12493) Checkpoint target option: remove feature (https://github.com/wiredtiger/wiredtiger/pull/10934)

**Review by korteland — COMMENTED** [review]
There's a couple of things missing from the description:
* WT-12470
* WT-12459
* WT-12460
* WT-12462

**Review by luke-pearson — APPROVED** [review]
LGTM, but Will's comment needs addressing.

**Comment by Sean04** [comment]
Thanks for double checking @korteland 
- [WT-12470](https://jira.mongodb.org/browse/WT-12470) is already in the description
- As for - [WT-12459](https://jira.mongodb.org/browse/WT-12459), [WT-12460](https://jira.mongodb.org/browse/WT-12460), [WT-12462](https://jira.mongodb.org/browse/WT-12462)  they look to have been merged into develop but then reverted and applied back to this branch in [WT-13172](https://jira.mongodb.org/browse/WT-13172) as reverts in the commit messages. Each line in the description is a copy of the commit message but I will edit these so that the ticket number comes first for clarity.

**Comment by korteland** [comment]
Thanks! Not sure how I missed WT-12470, probably a typo while searching.

The others would ideally be their own entry, since that's clearest - WT-13172 exists for paperwork reasons and isn't useful to anyone looking at what this commit actually removes.

**Review by korteland — APPROVED** [review]
LGTM! Nice work wrangling this big old thing into shape.

**Pull request merged**